src/lpython/parser/semantics.h

```diff
@@ -803,6 +803,15 @@ char* unescape(Allocator &al, LCompilers::Str &s) {
         } else if (s.p[idx] == '\\' && s.p[idx+1] == '\'') {
             x += "'";
             idx++;
+        } else if (s.p[idx] == '\\' && s.p[idx+1] == '\"') {
```

Comment:
Could you recheck for this case -

A tricky one I came across while randomly trying out a bunch of strings. Although a very improbable input, I would like this pr to perfectly handle all cases.
```
print('I said \'\\\'hi\\''hey\' ')

(lp) C:\Users\kunni\lpython>python try.py
I said '\'hi\hey'

(lp) C:\Users\kunni\lpython>src\bin\lpython try.py
I said '\'hi\hey\'
```

Thanks !

Reply:
I have observed that LPython generates incorrect output for the following cases:
```
print('  ''hey\' ')
print(" " " hey\" ")
```
Meaning, when there are 2 strings in a print function, which have no comma or any operator between them. This is accepted behavior in python.
Like you said this is an impossible input, since a user should put an operator like a comma between two strings, in which case lpython executes correctly.
I do not yet understand why lpython outputs that escape sequence after 'hey', but will try after my classes end. But right now I think that this might be a more parser related issue.

Reply:
> But right now I think that this might be a more parser related issue.

Try checking the AST using `--show-ast` command line argument in `lpython`.

Reply:
I was kinda right - lpython considers these both of the print statements I wrote as single strings and ignores the quotes in the middle, which somehow leads to the problem of printing the escape sequence.
I believe that in  ("a" " b") or ('a' 'b'), a and b should be considered two different strings by lpython, which it does not.  
Some AST results for further investigations:
```
print('a  b\' ')
[(ConstantStr "a  b' " ())]                           // Correct

print('a'' b\' ')
[(ConstantStr "a b\' " ())]                           // Incorrect (Same AST but '\' also printed)

print('a'              ' b\' ')
[(ConstantStr "a b\' " ())]                           // Incorrect (Whitespace ignored) 

print('a' , ' b\' ')
[(ConstantStr "a" ()) (ConstantStr " b' " ())]        // Correct (Two strings because of comma) 

print("a " " b\' ")
[(ConstantStr "a  b\' " ())]                          // Incorrect (Same issue with double quotes)

print("a' ' b\' ")
[(ConstantStr "a' ' b' " ())]                         // Correct (Nested quotes work)
```
I  think we should open a new issue for this, if it is a problem that needs to be solved.

Reply:
Yup, thanks for investigating. You can create a new issue or directly open a new PR and work towards fixing it.
If you get any doubts regarding the parser, do ping me I can help you.